internal/tofu/node_resource_plan.go

```diff
@@ -315,25 +338,12 @@ func (n *nodeExpandPlannableResource) resourceInstanceSubgraph(ctx EvalContext,
 
 		// If we're in legacy import mode (the import CLI command), we only need
 		// to return the import node, not a plannable resource node.
-		if n.legacyImportMode {
-			for _, importTarget := range n.importTargets {
-				if importTarget.Addr.Equal(a.Addr) {
-
-					// The import ID was supplied as a string on the command
-					// line and made into a synthetic HCL expression.
-					importId, diags := evaluateImportIdExpression(importTarget.ID, ctx)
-					if diags.HasErrors() {
-						// This should be impossible, because the import command
-						// arg parsing builds the synth expression from a
-						// non-null string.
-						panic(fmt.Sprintf("Invalid import id: %s. This is a bug in OpenTofu; please report it!", diags.Err()))
-					}
-
-					return &graphNodeImportState{
-						Addr:             importTarget.Addr,
-						ID:               importId,
-						ResolvedProvider: n.ResolvedProvider,
-					}
+		for _, c := range commandLineImportTargets {
```

Comment:
The comment still matches with the logic:
If there are import targets originating from an `tofu import` call, then these should be address by creating a `graphNodeImportState` node instead of a `NodePlannableResourceInstance`.
And technically, import targets either all originate from `tofu import` (`CommandLineImportTarget`) or all of them originate from `import` blocks (`Config`). The code does already rely on that - [example here](https://github.com/opentofu/opentofu/blob/64f419e33548d4a243b60c14ffe37be70e3eaf34/internal/tofu/node_resource_abstract.go#L302-L309)
Maybe this is something that could be better captured with some different encapsulation

I will change the "legacy import" mode comment to just say "import CLI" instead, as we do not currently consider the `tofu import` command as "legacy"

Reply:
e8ed3ea2c02d4540de5673882d74d510bf90e439 - Changed the comment a bit, removing the "legacy" terminology from "import CLI"